File: Manali/Week1/BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices)==1: return 0
        maximum = 0
        i = 0
        j = 1
        while j<len(prices) :
            if prices[j]-prices[i]<0:
                i=j
                j+=1
            else:
                maximum = max(maximum,prices[j]-prices[i])
                j+=1
        return maximum

File: parth/Week1/BestTimeToBuyAndSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int lowprice = prices[0];
        for(int i=1; i<prices.size(); i++)
        {
            if(prices[i] < lowprice)
            {
                lowprice = prices[i];
                continue;
            }
            profit = std::max(profit, prices[i] - lowprice);
        }
        return profit;
    }
};